pallets/fruniques/src/lib.rs

```diff
@@ -280,11 +282,19 @@ pub mod pallet {
 			attributes: Option<Attributes<T>>,
 			parent_info_call: Option<ParentInfoCall<T>>,
 		) -> DispatchResult {
+			//Ensure the collection exists
 			ensure!(Self::collection_exists(&class_id), Error::<T>::CollectionNotFound);
+			// Ensure the user is in the collection
 			let user: T::AccountId = ensure_signed(origin.clone())?;
-			Self::is_authorized(user.clone(), class_id, Permission::Mint)?;
+			ensure!(
```

Comment:
Instead of using an ensure with a validation check, I'll suggest using the `has_role`function from the rbac pallet. Let me know if works.

Reply:
why is _has_role_ a better choice for checking a particular permission? it seems that the only difference is that has_role checks if a user has any of a given set of permissions on the pallet
should i change is_authorized for has_role on the rest of the pallet(s) as well?